Feature : Forms creating using local LLM

ashu01304 opened the pull request:
I have attempted to integrate a local LLM using Ollama to generate forms through prompts. I would appreciate your review of the current implementation and any suggestions for improvements. 

https://github.com/user-attachments/assets/6699d344-1f76-44e7-9276-29a329cacd23

abh3po commented:
can we add a field in the settings somewhere to take the url as a parameter? My models run from a self hosted server. Which makes it difficult for me to test this.

But good work so far!

ashu01304 commented:
Add LLM settings management and integrate with local storage
- Implemented input fields for LLM Server URL and Model Name in FormSettings component.
- Added functionality to save and retrieve settings from local storage.
- Updated Ollama utility functions to use stored settings for API calls.

https://github.com/user-attachments/assets/845f90a5-84a6-45cb-b066-ff8c84383091

ashu01304 commented:
Please review this whenever you have time, and let me know if it needs some more modifications.

abh3po commented on packages/formstr-app/src/containers/CreateFormNew/components/FormSettings/index.tsx:
```diff
@@ -25,6 +27,40 @@ function FormSettings() {
     });
   };
 
+  const [ollamaUrlInput, setOllamaUrlInput] = useState<string>('');
```
should these go in the create form provider instead of states in the component?  not a request, but worth considering? Also should these settings only show up in the prompt screen?

ashu01304 replied:
You're right, it makes more sense for the AI settings to be shown only on the prompt screen. I'll make that change.

abh3po commented on packages/formstr-app/src/containers/CreateFormNew/components/LLMFormGenerator/index.tsx:
```diff
@@ -0,0 +1,213 @@
+        setIsLoading(true); 
+
+        const prompt = `
+You are an assistant that generates JSON structures for web forms based on user descriptions.
```
we can move the prompt to another file.

ashu01304 replied:
will do that asap.

abh3po commented on packages/formstr-app/src/containers/CreateFormNew/components/QuestionsList/index.tsx:
```diff
@@ -1,144 +1,261 @@
+                const options = Array.isArray(aiField.options) ? aiField.options : [];
+
+                const renderElement: AnswerTypes =
+                    fieldType === 'ShortText' ? AnswerTypes.shortText :
```
this could also have been an object, or a seperate function

ashu01304 replied:
Sure, I’ll make the necessary modifications.

abh3po commented on packages/formstr-app/src/utils/ollama.ts:
```diff
@@ -0,0 +1,82 @@
+/// <reference lib="es2015.promise" />
```
should we have our own `ollama.ts` are there libraries that do this?

abh3po replied:
also should we use ollama tools instead of a prompt? again, no right answers, just want to evaluate? https://ollama.com/blog/tool-support

ashu01304 replied:
actually, I wrote the custom fetch logic in ollama.ts mainly to keep things simple and dependency-free for the initial version. and yes, libraries  do exist, like ollama-js, which is an official TypeScript SDK for interacting with local Ollama models, we can definitely explore it.
Thanks for pointing out the Tool Support feature, I haven’t explored it in detail yet, i will definitely explore more about it, that could possibly make implementing things easier.

ashu01304 commented on packages/formstr-app/src/containers/CreateFormNew/components/LLMFormGenerator/index.tsx:
```diff
@@ -0,0 +1,213 @@
+                style={styles.textarea}
+                disabled={isLoading} 
+            />
+            <button
```
okay

abh3po commented:
Resolving the CORS is important for productionizing it. Will review the changes in a bit.
